### PR TITLE
i#6238: deprecate dr_fp_type_t structure and migrate to dr_instr_category_t

### PR DESCRIPTION
**Pull request by kuhanov**
Continue #6331 discussion
FP types are not needed now because all information is in category now

Issue: https://github.com/DynamoRIO/dynamorio/issues/6238

### CODE REVIEW

File: core/ir/instr_api.h [hunk 1]
```diff
@@ -1928,6 +1928,16 @@ DR_API
  *   non-NULL, the type of the floating point operation is written to \p type.
  */
 bool
+instr_is_floating_exten(instr_t *instr, dr_instr_category_t *type);
```
Comment (derekbruening) [inline core/ir/instr_api.h:1931]: Not sure about the name "_exten": how about `instr_is_floating_type` or `instr_get_floating_type`

File: core/ir/x86/instr.c [hunk 2]
```diff
@@ -907,6 +907,17 @@ instr_is_rep_string_op(instr_t *instr)
...
+instr_is_floating_exten(instr_t *instr, dr_instr_category_t *type OUT)
+{
+    uint cat = instr_get_category(instr);
+    if (!TEST(DR_INSTR_CATEGORY_FP, cat))
+        return false;
+    if (type != NULL)
+        *type = cat;
+    return true;
+}
+
```
Comment (derekbruening) [inline core/ir/x86/instr.c:920]: Also change instr_is_floating() to call this one instead of the deprecated one?

File: core/ir/instr_api.h (hunk 1 above)
Comment (kuhanov) [reply core/ir/instr_api.h:1931]: instr_is_floating_type

File: core/ir/x86/instr.c (hunk 2 above)
Comment (kuhanov) [reply core/ir/x86/instr.c:920]: Sure, missed that

File: core/ir/instr_api.h [hunk 3]
```diff
@@ -1911,7 +1911,7 @@ typedef enum {
 } dr_instr_category_t;
 
 /**
- * Indicates which type of floating-point operation and instruction performs.
+ * \deprecated Indicates which type of floating-point operation and instruction performs.
```
[the remark this answers is not in the document]
Comment (kuhanov) [reply core/ir/instr_api.h:1914]: Thank you for pointing me in the right direction.

File: api/docs/release.dox [hunk 4]
```diff
@@ -149,6 +149,10 @@ Further non-compatibility-affecting changes include:
  - Added type_is_read() API that returns true if a trace type reads from memory.
  - Added instr_num_memory_read_access() and instr_num_memory_write_access() that return
    the number of memory read and write accesses of an instruction respectively.
+ - Deprecated #dr_fp_type_t for Floating-Point operation types amd migrate to
+   #dr_instr_category_t.
+   Deprecated instr_is_floating_ex(), replacing them with instr_is_floating_exten().
```
Comment (derekbruening) [inline api/docs/release.dox:154]: Stale name instr_is_floating_exten() -- surprised Doxygen doesn't fail the build (parens trigger a link -- maybe not as mandatory as a # link so it silently skips?).

Comment (derekbruening) [inline api/docs/release.dox:154]: grammar nit: s/them/it/

File: api/docs/release.dox [hunk 5]
```diff
@@ -149,6 +149,10 @@ Further non-compatibility-affecting changes include:
  - Added type_is_read() API that returns true if a trace type reads from memory.
  - Added instr_num_memory_read_access() and instr_num_memory_write_access() that return
    the number of memory read and write accesses of an instruction respectively.
+ - Deprecated #dr_fp_type_t for Floating-Point operation types amd migrate to
```
Comment (derekbruening) [inline api/docs/release.dox:152]: spelling and grammar: "amd migrate".  I would say sthg like "...types in favor of the new #dr_instr_category_t."

**Review by derekbruening — APPROVED** [review]

File: api/docs/release.dox (hunk 5 above)
Comment (kuhanov) [reply api/docs/release.dox:152]: ok.

**Pull request merged**